shipping_addresses_test.go

```diff
@@ -0,0 +1,255 @@
+			  </shipping_address>
+			  <shipping_address href="https://your-subdomain.recurly.com/v2/accounts/1/shipping_addresses/2">
+			    <account href="https://your-subdomain.recurly.com/v2/accounts/1"/>
+			    <shipping_address_id>2</shipping_address_id>
+			    <nickname>Home</nickname>
+			    <first_name>Verena</first_name>
+			    <last_name>Example</last_name>
+			    <phone>555-867-5309</phone>
+			    <email>verena@example.com</email>
+			    <address1>123 Fourth St.</address1>
+			    <address2>Apt. 101</address2>
+			    <city>San Francisco</city>
+			    <state>CA</state>
+			    <zip>94105</zip>
+			    <country>US</country>
+			    <created_at type="datetime">2018-03-19T15:48:00Z</created_at>
+			    <updated_at type="datetime">2018-03-19T15:48:00Z</updated_at>
+			  </shipping_address>
+			</shipping_addresses>`)
+	})
+	r, shippingAddresses, err := client.ShippingAddresses.ListAccount("1", recurly.Params{"per_page": 1})
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	} else if r.IsError() {
+		t.Fatal("expected list subcriptions to return OK")
+	} else if pp := r.Request.URL.Query().Get("per_page"); pp != "1" {
+		t.Fatalf("unexpected per_page: %s", pp)
+	}
+
+	created, _ := time.Parse(recurly.DateTimeFormat, "2018-03-19T15:48:00Z")
+	updated, _ := time.Parse(recurly.DateTimeFormat, "2018-03-19T15:48:00Z")
+
+	if diff := cmp.Diff(shippingAddresses, []recurly.ShippingAddress{
+		{
+			ID:          "2438622711411416831",
+			AccountCode: "1",
+			Nickname:    "Work",
+			FirstName:   "Verena",
+			LastName:    "Example",
+			Company:     "Recurly Inc",
+			Email:       "verena@example.com",
+			Address:     "123 Main St.",
+			Address2:    "Suite 101",
+			City:        "San Francisco",
+			State:       "CA",
+			Zip:         "94105",
+			Country:     "US",
+			Phone:       "555-222-1212",
+			CreatedAt:   recurly.NewTime(created),
+			UpdatedAt:   recurly.NewTime(updated),
+		},
+		{
+			AccountCode: "1",
+			FirstName:   "Verena",
+			LastName:    "Example",
+			Nickname:    "Home",
+			Email:       "verena@example.com",
+			Address:     "123 Fourth St.",
+			Address2:    "Apt. 101",
+			City:        "San Francisco",
+			State:       "CA",
+			Zip:         "94105",
+			Country:     "US",
+			Phone:       "555-867-5309",
+			CreatedAt:   recurly.NewTime(created),
+			UpdatedAt:   recurly.NewTime(updated),
+		},
+	}); diff != "" {
+		t.Fatal(diff)
+	}
+}
+
+func TestShippingAddress_Create(t *testing.T) {
+	setup()
+	defer teardown()
+	mux.HandleFunc("/v2/accounts/1/shipping_addresses", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			t.Fatalf("unexpected method: %s", r.Method)
+		}
+		w.WriteHeader(201)
+		fmt.Fprint(w, `<?xml version="1.0" encoding="UTF-8"?><shipping_address href="https://your-subdomain.recurly.com/v2/accounts/1/shipping_addresses/2438622711411416831"></shipping_address>`)
+	})
+
+	r, _, err := client.ShippingAddresses.Create("1", recurly.ShippingAddress{})
```

Comment:
Please add a few `shipping_address` fields in the XML and assert the second (currently discarded) value with `cmp.Diff` to ensure it's being populated with the XML from the response. There's currently a bug here.